WAF IpSet broken with IPV6 for Cloudfront Distribution on v5.35

### What happened?

After bumping `pulumi_aws` to 5.36 (from 5.34) our `aws.wafv2.IpSet` crashes on `preview` command.


### Expected Behavior

Creation of `IpSet` resource.

### Steps to reproduce

Creation of Pulumi stack with:

```
aws.wafv2.IpSet(
        "ip6_sample",
        addresses=["2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"], # Random IPV6
        description="Step to reproduce issue",
        ip_address_version="IPV6",
        scope="CLOUDFRONT",
        opts=pulumi.ResourceOptions(
            provider=aws.Provider(
                "us-east-1",
                region="us-east-1"
            ),
        )
)
```

### Output of `pulumi preview`

```
panic: interface conversion: interface {} is nil, not map[string]interface {}
    goroutine 76 [running]:
    github.com/hashicorp/go-cty/cty.Value.GetAttr({{{0xef06d38?, 0xc002fa31b0?}}, {0x0?, 0x0?}}, {0xd9d6707, 0x9})
    	/home/runner/go/pkg/mod/github.com/hashicorp/go-cty@v1.4.1-0.20200414143053-d3edf31b6320/cty/value_ops.go:711 +0x2ed
    github.com/hashicorp/terraform-provider-aws/internal/service/wafv2.ResourceIPSet.func2({0x0?, 0xd9d3250?}, {0xef05ad8?, 0xc000136008?}, {0xd9d6707?, 0x9?}, 0xc0015e7040?)
    	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/wafv2/ip_set.go:61 +0x4b
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x6?, {0xef05ad8, 0xc000136008}, {0xd9d6707, 0x9}, 0xc0015e7040, 0xc003318780, {0xef13eb0?, 0xc003318800}, 0x0)
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230327102345-3fa930f86570/helper/schema/schema.go:1144 +0x3a8
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc0015963c0, {0xef05ad8, 0xc000136008}, 0xc0032f7380, 0xc003312bd0, 0xc000886c00, {0xd986400, 0xc0004e4000}, 0x0)
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230327102345-3fa930f86570/helper/schema/schema.go:679 +0x353
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0xc0014615f0?, {0xef05ad8?, 0xc000136008?}, 0xc0032f7380, 0x8010105?, {0xd986400?, 0xc0004e4000?})
    	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20230327102345-3fa930f86570/helper/schema/resource.go:895 +0x6c
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.simpleDiff({0x7f6ed7280108?, {0x30?, 0x178bd880?, 0xc003312bd0?}}, 0x0?, 0xbc9c9c0?, 0xc0033e9400?, 0x40d907?, {{{0xef06d38, 0xc002fa2d80}}, ...}, ...)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.0/pkg/tfshim/sdk-v2/provider_diff.go:73 +0x66c
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.Diff({0xc00075db00?, {0x0?, 0x0?, 0x0?}}, {0xda17ec7, 0x10}, {0xef071d0?, 0xc0033208f0}, {0xee3bbc0, 0xc003312bd0})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.0/pkg/tfshim/sdk-v2/provider_diff.go:54 +0x2b6
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Diff(0xc002bb8000, {0xef05b48?, 0xc003312720?}, 0xc001cb7ce0)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.0/pkg/tfbridge/provider.go:785 +0x571
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Diff_Handler.func1({0xef05b48, 0xc003312720}, {0xd337260?, 0xc001cb7ce0})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.59.0/proto/go/provider_grpc.pb.go:555 +0x78
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0xef05b48, 0xc0033125a0}, {0xd337260, 0xc001cb7ce0}, 0xc0032e2cc0, 0xc0032d54a0)
    	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3f9
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Diff_Handler({0xd80f2a0?, 0xc002bb8000}, {0xef05b48, 0xc0033125a0}, 0xc001cb7c70, 0xc0021d86c0)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.59.0/proto/go/provider_grpc.pb.go:557 +0x138
    google.golang.org/grpc.(*Server).processUnaryRPC(0xc000866000, {0xef16f20, 0xc002cc16c0}, 0xc00331a120, 0xc002b8e720, 0x1789fa88, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
    google.golang.org/grpc.(*Server).handleStream(0xc000866000, {0xef16f20, 0xc002cc16c0}, 0xc00331a120, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
    google.golang.org/grpc.(*Server).serveStreams.func1.2()
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0x98
    created by google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

  aws:wafv2:IpSet (CloudflareIPV6):
    error: error reading from server: EOF

```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/tests/regress_1020_test.go
+++ b/pkg/tests/regress_1020_test.go
@@ -16,7 +16,6 @@ package tests
 
 import (
 	"context"
-	"fmt"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -63,9 +62,6 @@ func TestRegress1020(t *testing.T) {
 				MaxItems: 10000,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-					if d.GetRawPlan().IsNull() {
-						panic(fmt.Sprintf("%s", "NULL GetRawPlan"))
-					}
 					if d.GetRawPlan().GetAttr("addresses").IsWhollyKnown() {
 						o, n := d.GetChange("addresses")
 						oldAddresses := o.(*schema.Set).List()
@@ -213,15 +209,10 @@ func TestRegress1020(t *testing.T) {
 			"1.2.3.4/32",
 			"5.6.7.8/32"
 		      ],
-		      "arn": "arn:aws:wafv2:us-west-2:616138583583:regional/ipset/ip6_sample-6d6da96/f25bceeb-022a-4330-95d2-b76c7729ed61",
-		      "description": "Step to reproduce issue",
 		      "id": "f25bceeb-022a-4330-95d2-b76c7729ed61",
 		      "ipAddressVersion": "IPV4",
-		      "lockToken": "d282421a-16a6-48b7-a27e-20731d445ab2",
 		      "name": "ip6_sample-6d6da96",
-		      "scope": "REGIONAL",
-		      "tags": {},
-		      "tagsAll": {}
+		      "scope": "REGIONAL"
 		    },
 		    "news": {
 		      "__defaults": [
@@ -231,13 +222,20 @@ func TestRegress1020(t *testing.T) {
 			"1.2.3.4/32",
 			"5.6.7.9/32"
 		      ],
-		      "description": "Step to reproduce issue",
 		      "ipAddressVersion": "IPV4",
 		      "name": "ip6_sample-6d6da96",
 		      "scope": "REGIONAL"
 		    }
 		  },
-		  "response": {}
+		  "response": {
+                    "changes": "DIFF_SOME",
+                    "stables": ["name", "scope", "ipAddressVersion"],
+                    "hasDetailedDiff": true,
+                    "diffs": ["addresses"],
+                    "detailedDiff": {
+                       "addresses[1]": {"kind": "UPDATE"}
+                    }
+                  }
 		}`
 		testutils.Replay(t, server, testCase)
 	})

--- a/pkg/tests/regress_1020_test.go
+++ b/pkg/tests/regress_1020_test.go
@@ -1,0 +1,191 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	testutils "github.com/pulumi/pulumi-terraform-bridge/testing/x"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"net"
+)
+
+// See https://github.com/pulumi/pulumi-terraform-bridge/issues/1020
+func TestRegress1020(t *testing.T) {
+	ctx := context.Background()
+
+	CIDRBlocksEqual := func(cidr1, cidr2 string) bool {
+		ip1, ipnet1, err := net.ParseCIDR(cidr1)
+		if err != nil {
+			return false
+		}
+		ip2, ipnet2, err := net.ParseCIDR(cidr2)
+		if err != nil {
+			return false
+		}
+
+		return ip2.String() == ip1.String() && ipnet2.String() == ipnet1.String()
+	}
+
+	emptySchema := func() *schema.Schema {
+		return &schema.Schema{
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{},
+			},
+		}
+	}
+
+	resource := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 10000,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.GetRawPlan().IsNull() {
+						panic(fmt.Sprintf("%s", "NULL GetRawPlan"))
+					}
+					if d.GetRawPlan().GetAttr("addresses").IsWhollyKnown() {
+						o, n := d.GetChange("addresses")
+						oldAddresses := o.(*schema.Set).List()
+						newAddresses := n.(*schema.Set).List()
+						if len(oldAddresses) == len(newAddresses) {
+							for _, ov := range oldAddresses {
+								hasAddress := false
+								for _, nv := range newAddresses {
+									if CIDRBlocksEqual(ov.(string), nv.(string)) {
+										hasAddress = true
+										break
+									}
+								}
+								if !hasAddress {
+									return false
+								}
+							}
+							return true
+						}
+					}
+					return false
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_address_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"override": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"count": emptySchema(),
+									"none":  emptySchema(),
+								},
+							},
+						},
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tfProvider := &schema.Provider{
+		Schema: map[string]*schema.Schema{},
+		ResourcesMap: map[string]*schema.Resource{
+			"aws_wafv2_ip_set": resource,
+		},
+	}
+
+	p := shimv2.NewProvider(tfProvider) // , shimv2.WithDiffStrategy(shimv2.PlanState))
+
+	info := tfbridge.ProviderInfo{
+		P:           p,
+		Name:        "aws",
+		Description: "A Pulumi package for creating and managing Amazon Web Services (AWS) cloud resources.",
+		Keywords:    []string{"pulumi", "aws"},
+		License:     "Apache-2.0",
+		Homepage:    "https://pulumi.io",
+		Repository:  "https://github.com/pulumi/pulumi-aws",
+		Version:     "0.0.2",
+		Resources: map[string]*tfbridge.ResourceInfo{
+			"aws_wafv2_ip_set": {Tok: "aws:wafv2/ipSet:IpSet"},
+		},
+	}
+
+	server := tfbridge.NewProvider(ctx,
+		nil,      /* hostClient */
+		"aws",    /* module */
+		"",       /* version */
+		p,        /* tf */
+		info,     /* info */
+		[]byte{}, /* pulumiSchema */
+	)
+
+	testCase := `
+{
+  "method": "/pulumirpc.ResourceProvider/Create",
+  "request": {
+    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
+    "properties": {
+      "__defaults": [
+        "name"
+      ],
+      "addresses": [
+        "2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
+      ],
+      "ipAddressVersion": "IPV6",
+      "name": "ip6_sample-e8442ad",
+      "scope": "CLOUDFRONT"
+    },
+    "preview": true
+  },
+  "response": {}
+}`
+	testutils.Replay(t, server, testCase)
+}

--- a/pkg/tests/regress_1020_test.go
+++ b/pkg/tests/regress_1020_test.go
@@ -167,35 +167,78 @@ func TestRegress1020(t *testing.T) {
 		[]byte{}, /* pulumiSchema */
 	)
 
-	testCase := `
-	{
-	  "method": "/pulumirpc.ResourceProvider/Create",
-	  "request": {
-	    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
-	    "properties": {
-	      "__defaults": [
-		"name"
-	      ],
-	      "addresses": [
-		"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
-	      ],
-	      "ipAddressVersion": "IPV6",
-	      "name": "ip6_sample-e8442ad",
-	      "scope": "CLOUDFRONT"
-	    },
-	    "preview": true
-	  },
-	  "response": {
-	    "properties": {
-	      "addresses": [
-		"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
-	       ],
-	       "id": "",
-	       "ipAddressVersion": "IPV6",
-	       "name": "ip6_sample-e8442ad",
-	       "scope": "CLOUDFRONT"
-	    }
-	  }
-	}`
-	testutils.Replay(t, server, testCase)
+	t.Run("can preview Create", func(t *testing.T) {
+		testCase := `
+		{
+		  "method": "/pulumirpc.ResourceProvider/Create",
+		  "request": {
+		    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
+		    "properties": {
+		      "__defaults": [
+			"name"
+		      ],
+		      "addresses": [
+			"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
+		      ],
+		      "ipAddressVersion": "IPV6",
+		      "name": "ip6_sample-e8442ad",
+		      "scope": "CLOUDFRONT"
+		    },
+		    "preview": true
+		  },
+		  "response": {
+		    "properties": {
+		      "addresses": [
+			"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
+		       ],
+		       "id": "",
+		       "ipAddressVersion": "IPV6",
+		       "name": "ip6_sample-e8442ad",
+		       "scope": "CLOUDFRONT"
+		    }
+		  }
+		}`
+		testutils.Replay(t, server, testCase)
+	})
+
+	t.Run("can compute an Update plan in Diff", func(t *testing.T) {
+		testCase := `
+		{
+		  "method": "/pulumirpc.ResourceProvider/Diff",
+		  "request": {
+		    "id": "f25bceeb-022a-4330-95d2-b76c7729ed61",
+		    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
+		    "olds": {
+		      "addresses": [
+			"1.2.3.4/32",
+			"5.6.7.8/32"
+		      ],
+		      "arn": "arn:aws:wafv2:us-west-2:616138583583:regional/ipset/ip6_sample-6d6da96/f25bceeb-022a-4330-95d2-b76c7729ed61",
+		      "description": "Step to reproduce issue",
+		      "id": "f25bceeb-022a-4330-95d2-b76c7729ed61",
+		      "ipAddressVersion": "IPV4",
+		      "lockToken": "d282421a-16a6-48b7-a27e-20731d445ab2",
+		      "name": "ip6_sample-6d6da96",
+		      "scope": "REGIONAL",
+		      "tags": {},
+		      "tagsAll": {}
+		    },
+		    "news": {
+		      "__defaults": [
+			"name"
+		      ],
+		      "addresses": [
+			"1.2.3.4/32",
+			"5.6.7.9/32"
+		      ],
+		      "description": "Step to reproduce issue",
+		      "ipAddressVersion": "IPV4",
+		      "name": "ip6_sample-6d6da96",
+		      "scope": "REGIONAL"
+		    }
+		  },
+		  "response": {}
+		}`
+		testutils.Replay(t, server, testCase)
+	})
 }

--- a/pkg/tests/regress_1020_test.go
+++ b/pkg/tests/regress_1020_test.go
@@ -168,24 +168,34 @@ func TestRegress1020(t *testing.T) {
 	)
 
 	testCase := `
-{
-  "method": "/pulumirpc.ResourceProvider/Create",
-  "request": {
-    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
-    "properties": {
-      "__defaults": [
-        "name"
-      ],
-      "addresses": [
-        "2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
-      ],
-      "ipAddressVersion": "IPV6",
-      "name": "ip6_sample-e8442ad",
-      "scope": "CLOUDFRONT"
-    },
-    "preview": true
-  },
-  "response": {}
-}`
+	{
+	  "method": "/pulumirpc.ResourceProvider/Create",
+	  "request": {
+	    "urn": "urn:pulumi:dev::repro-1020::aws:wafv2/ipSet:IpSet::ip6_sample",
+	    "properties": {
+	      "__defaults": [
+		"name"
+	      ],
+	      "addresses": [
+		"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
+	      ],
+	      "ipAddressVersion": "IPV6",
+	      "name": "ip6_sample-e8442ad",
+	      "scope": "CLOUDFRONT"
+	    },
+	    "preview": true
+	  },
+	  "response": {
+	    "properties": {
+	      "addresses": [
+		"2001:0db8:85a3:0000:0000:8a2e:0370:7334/32"
+	       ],
+	       "id": "",
+	       "ipAddressVersion": "IPV6",
+	       "name": "ip6_sample-e8442ad",
+	       "scope": "CLOUDFRONT"
+	    }
+	  }
+	}`
 	testutils.Replay(t, server, testCase)
 }

--- a/pkg/tests/regress_1020_test.go
+++ b/pkg/tests/regress_1020_test.go
@@ -229,7 +229,7 @@ func TestRegress1020(t *testing.T) {
 		  },
 		  "response": {
                     "changes": "DIFF_SOME",
-                    "stables": ["name", "scope", "ipAddressVersion"],
+                    "stables": "*",
                     "hasDetailedDiff": true,
                     "diffs": ["addresses"],
                     "detailedDiff": {

--- a/pkg/tfshim/sdk-v2/provider_diff.go
+++ b/pkg/tfshim/sdk-v2/provider_diff.go
@@ -43,14 +43,27 @@ func (p v2Provider) Diff(t string, s shim.InstanceState, c shim.ResourceConfig) 
 
 	config, state := configFromShim(c), stateFromShim(s)
 	rawConfig := makeResourceRawConfig(opts.diffStrategy, config, r)
-	if state != nil {
+
+	if state == nil {
+		// When handling Create Pulumi passes nil for state, but this diverges from how Terraform does things,
+		// see: https://github.com/pulumi/pulumi-terraform-bridge/issues/911 and can lead to panics. Compensate
+		// by constructing an InstanceState.
+
+		state = &terraform.InstanceState{
+			RawState:  r.CoreConfigSchema().EmptyValue(),
+			RawConfig: rawConfig,
+			RawPlan:   rawConfig,
+		}
+	} else {
 		state.RawConfig = rawConfig
+
+		// Upgrades are needed only if we have non-empty prior state.
+		state, err = upgradeResourceState(p.tf, r, state)
+		if err != nil {
+			return nil, fmt.Errorf("failed to upgrade resource state: %w", err)
+		}
 	}
 
-	state, err = upgradeResourceState(p.tf, r, state)
-	if err != nil {
-		return nil, fmt.Errorf("failed to upgrade resource state: %w", err)
-	}
 	diff, err := p.simpleDiff(opts.diffStrategy, r, state, config, rawConfig, p.tf.Meta())
 	if diff != nil {
 		diff.RawConfig = rawConfig
@@ -127,6 +140,7 @@ func simpleDiffViaPlanState(
 	if err != nil {
 		return nil, err
 	}
+
 	planned := terraform.NewResourceConfigShimmed(proposedNewStateVal, res.CoreConfigSchema())
 	return res.SimpleDiff(ctx, s, planned, meta)
 }

--- a/pkg/tfshim/sdk-v2/provider_diff.go
+++ b/pkg/tfshim/sdk-v2/provider_diff.go
@@ -52,11 +52,8 @@ func (p v2Provider) Diff(t string, s shim.InstanceState, c shim.ResourceConfig) 
 		state = &terraform.InstanceState{
 			RawState:  r.CoreConfigSchema().EmptyValue(),
 			RawConfig: rawConfig,
-			RawPlan:   rawConfig,
 		}
 	} else {
-		state.RawConfig = rawConfig
-
 		// Upgrades are needed only if we have non-empty prior state.
 		state, err = upgradeResourceState(p.tf, r, state)
 		if err != nil {
@@ -83,6 +80,11 @@ func (p v2Provider) simpleDiff(
 
 	switch diffStrat {
 	case ClassicDiff:
+		if s.RawPlan.IsNull() {
+			sCopy := s.DeepCopy()
+			sCopy.RawPlan = rawConfigVal
+			return res.SimpleDiff(ctx, sCopy, c, meta)
+		}
 		return res.SimpleDiff(ctx, s, c, meta)
 	case PlanState:
 		return simpleDiffViaPlanState(ctx, res, s, rawConfigVal, meta)

--- a/pkg/tfshim/sdk-v2/provider_diff.go
+++ b/pkg/tfshim/sdk-v2/provider_diff.go
@@ -50,7 +50,6 @@ func (p v2Provider) Diff(t string, s shim.InstanceState, c shim.ResourceConfig) 
 		// by constructing an InstanceState.
 
 		state = &terraform.InstanceState{
-			RawState:  r.CoreConfigSchema().EmptyValue(),
 			RawConfig: rawConfig,
 		}
 	} else {
@@ -81,9 +80,11 @@ func (p v2Provider) simpleDiff(
 	switch diffStrat {
 	case ClassicDiff:
 		if s.RawPlan.IsNull() {
-			sCopy := s.DeepCopy()
-			sCopy.RawPlan = rawConfigVal
-			return res.SimpleDiff(ctx, sCopy, c, meta)
+			// SimpleDiff may read RawPlan and panic if it is nil; while in the case of ClassicDiff we do
+			// not yet do what TF CLI does (that is PlanState), it is better to approximate and assume that
+			// the RawPlan is the same as RawConfig than to have the code panic.
+			rawPlan := rawConfigVal
+			return res.SimpleDiff(ctx, instanceStateWithUpdatedRawPlan(s, rawPlan), c, meta)
 		}
 		return res.SimpleDiff(ctx, s, c, meta)
 	case PlanState:
@@ -125,6 +126,15 @@ func (p v2Provider) simpleDiff(
 	default:
 		return res.SimpleDiff(ctx, s, c, meta)
 	}
+}
+
+func instanceStateWithUpdatedRawPlan(
+	s *terraform.InstanceState,
+	rawPlan hcty.Value,
+) *terraform.InstanceState {
+	c := s.DeepCopy()
+	c.RawPlan = rawPlan
+	return c
 }
 
 func simpleDiffViaPlanState(

